src/js/components/Button/Button.js

```diff
@@ -26,134 +27,129 @@ const isDarkBackground = props => {
   return colorIsDark(backgroundColor, props.theme);
 };
 
-class Button extends Component {
-  static defaultProps = {
-    type: 'button',
-    focusIndicator: true,
-    gap: 'small',
-  };
-
-  constructor(props) {
-    super(props);
-
-    const { children, icon, label } = props;
-    if ((icon || label) && children) {
-      console.warn(
-        'Button should not have children if icon or label is provided',
-      );
-    }
+const Button = props => {
+  const {
+    a11yTitle,
+    color, // munged to avoid styled-components putting it in the DOM
+    forwardRef,
+    children,
+    disabled,
+    icon,
+    gap,
+    fill, // munged to avoid styled-components putting it in the DOM
+    focus,
+    href,
+    label,
+    onClick,
+    onMouseOut,
+    onMouseOver,
+    plain,
+    primary,
+    reverse,
+    theme,
+    type,
+    as,
+    ...rest
+  } = props;
```

Comment:
can we move those in the props directly ? would be much easier to read.

Reply:
I will work on this and test to make sure everything is still working correctly! In Progress